### ISSUE TITLE
Weird Artifacts during Super Resolution of CAVE Dataset at x32 or higher

### ISSUE THREAD

**Issue by mylifeasazucchini** [JiangtaoNie/Rec_HSISR_PixAwaRefin#1]
Hi!

Amazing work and very clearly explained paper. However, in my experiments I noticed a few concerning artifacts which looked like to have a regular pattern. I am relatively new to this domain and struggle to understand these artifacts, can you suggest something to overcome these ? (The first image is from a simulated dataset and the second is from the commonly used CAVE dataset.)

![Picture1](https://user-images.githubusercontent.com/73758944/114902656-b488be00-9e1e-11eb-8974-a8ee0f0482af.png)
![Picture2](https://user-images.githubusercontent.com/73758944/114902663-b6528180-9e1e-11eb-82d1-11ccfb09b40f.png)

I also noticed that for higher scale factors these errors only get worse. Any help will be amazing! Thank you

**Comment by JiangtaoNie** [comment]
I think this problem may caused by the blur kernel in generating LR HSI is not identical with the one used in test phase. DIP based methods need to learn the exact degeneration model, or that may result in sever performance degradation.

**Comment by mylifeasazucchini** [comment]
Thank you for your quick response but would it be possible for you to share your script for generating the LR HSI ? 

I ask because following your advice, I tried to use the kernel generated by your get_kernel function in downsampler.py and made sure to have the same downsampler function in both data creation and trainig but those artifacts only became more obvious. 
For reference, this is how I am currently generating the LR HSI from the GT:

X = ---- the loaded GT hyperspectral Image
R = ----- spectral response of camera
C = ----- kernel from Downsampler.py's get kernel function (used bilinear setting for now)

Z = np.zeros([16,16,31])  # empty array for LR HSI
Y = np.tensordot(X,R,(2,0))  # computing the RGB

for j in range(32): # (32 because scale factor is 32)
                for k in range(32):
                    Z = Z + X[j:512:32,k:512:32,:]*C[k,j]

Is there something wrong in this formulation ? Thanks once again

**Comment by mylifeasazucchini** [comment]
Yes! I solved the issue, my kernels although they were correct, they were too wide coz I had failed to note the kernel width parameter. And after fixing it, I can say that those artifacts do not appear at all.

The fix I used was (for others who might have similar issue while generating training data)

ker = get_kernel(32,'gauss',0,8,support=None, sigma=3)
pad = abs(int((ker.shape[0] - 32) / 2.))
ker2=np.pad(ker, ((pad,pad),(pad,pad)), 'edge')

and I used this ker2 variable as the kernel in my data generation.

Thanks once again!
